File: go/delete-leaves-with-a-given-value.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 func RemoveLeaves(node *TreeNode, target int) *TreeNode {
    if node == nil {
        return nil
    }
    node.Left = RemoveLeaves(node.Left, target)
    node.Right = RemoveLeaves(node.Right, target)
    if node.Left == nil && node.Right == nil && node.Val == target {
        return nil
    }
    return node
}

func removeLeafNodes(root *TreeNode, target int) *TreeNode {
    return RemoveLeaves(root, target)
}

File: python/delete-leaves-with-a-given-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def delete_leaves(node: Optional[TreeNode], target: int) -> Optional[TreeNode]:
        if node == None:
            return None
        node.left = Solution.delete_leaves(node.left, target)
        node.right = Solution.delete_leaves(node.right, target)
        if node.left == None and node.right == None and node.val == target:
            return None
        return node

    def removeLeafNodes(self, root: Optional[TreeNode], target: int) -> Optional[TreeNode]:
        return Solution.delete_leaves(root, target)
